File: generated_code_examples/haskell/classification/random_forest.hs
module Model where
addVectors :: [Double] -> [Double] -> [Double]
addVectors v1 v2 = zipWith (+) v1 v2
mulVectorNumber :: [Double] -> Double -> [Double]
mulVectorNumber v1 num = [i * num | i <- v1]
score :: [Double] -> [Double]
score input =
    mulVectorNumber (addVectors (func0) (func1)) (0.5)
    where
        func0 =
            if (((input) !! (3)) <= (0.8))
                then
                    [1.0, 0.0, 0.0]
                else
                    if (((input) !! (2)) <= (4.8500004))
                        then
                            [0.0, 0.9622641509433962, 0.03773584905660377]
                        else
                            if (((input) !! (3)) <= (1.75))
                                then
                                    if (((input) !! (3)) <= (1.6500001))
                                        then
                                            [0.0, 0.25, 0.75]
                                        else
                                            [0.0, 1.0, 0.0]
                                else
                                    [0.0, 0.0, 1.0]
        func1 =
            if (((input) !! (3)) <= (0.8))
                then
                    [1.0, 0.0, 0.0]
                else
                    if (((input) !! (0)) <= (6.1499996))
                        then
                            if (((input) !! (2)) <= (4.8500004))
                                then
                                    [0.0, 0.9090909090909091, 0.09090909090909091]
                                else
                                    [0.0, 0.0, 1.0]
                        else
                            if (((input) !! (3)) <= (1.75))
                                then
                                    [0.0, 0.8666666666666667, 0.13333333333333333]
                                else
                                    [0.0, 0.0, 1.0]

File: generated_code_examples/elixir/classification/random_forest.ex
defmodule Model do
    @compile {:inline, read: 2}
    defp read(bin, pos) do
        <<_::size(pos)-unit(64)-binary, value::float, _::binary>> = bin
        value
    end
    defp list_to_binary(list) do
        for i <- list, into: <<>>, do: <<i::float>>
    end
    def score(input) do
        input = list_to_binary(input)
        func0 = fn ->
            cond do read(input,3) <= 0.75 ->
                [1.0, 0.0, 0.0]
            true ->
                cond do read(input,2) <= 4.75 ->
                    [0.0, 1.0, 0.0]
                true ->
                    cond do read(input,2) <= 5.049999952316284 ->
                        cond do read(input,3) <= 1.75 ->
                            [0.0, 0.8333333333333334, 0.16666666666666666]
                        true ->
                            [0.0, 0.08333333333333333, 0.9166666666666666]
                        end
                    true ->
                        [0.0, 0.0, 1.0]
                    end
                end
            end
        end
        func1 = fn ->
            cond do read(input,3) <= 0.800000011920929 ->
                [1.0, 0.0, 0.0]
            true ->
                cond do read(input,0) <= 6.25 ->
                    cond do read(input,2) <= 4.8500001430511475 ->
                        [0.0, 0.9487179487179487, 0.05128205128205128]
                    true ->
                        [0.0, 0.0, 1.0]
                    end
                true ->
                    cond do read(input,3) <= 1.550000011920929 ->
                        [0.0, 0.8333333333333334, 0.16666666666666666]
                    true ->
                        [0.0, 0.02564102564102564, 0.9743589743589743]
                    end
                end
            end
        end
        mul_vector_number(add_vectors(func0.(), func1.()), 0.5)
    end
defp add_vectors(v1_list, v2_list) do
  for {a,b} <- Enum.zip(v1_list, v2_list), do: a+b
end

defp mul_vector_number(v1, num) do
  for i <- v1, do: i * num
end
end
